pkg/volume/util/util.go

```diff
@@ -308,6 +318,115 @@ func LoadPodFromFile(filePath string) (*v1.Pod, error) {
 	return pod, nil
 }
 
+// PopulateZoneConfig populates the zone configurations based on several factors: node.zone; zone/zones from storageclass; allowedTopologies and Replicas
+// For single-zone volumes i.e. replicas = 1:
+//    VolumeZoneConfig.Zone, if present, should be used as the zone of the volume.
+//    VolumeZoneConfig.Zones, if present, should be used to choose a zone for a volume.
+//    VolumeZoneConfig.ReplicaZoneFromNode can be ignored
+//	  if neither Zone/Zones present, query and choose 1 zone from active zones
+// For multi-zone volumes i.e. replicas > 1:
+//    VolumeZoneConfig.ReplicaZoneFromNode, if present, should be used for one of the replica zones. Consider ZonesAlreadyPicked = 1
+//    VolumeZoneConfig.Zones, if present, should be used to choose N zones for the volume where N = replicas - ZonesAlreadyPicked
+//    if VolumeZoneConfig.Zones is absent, query and choose N zones from active zones where N = replicas - ZonesAlreadyPicked
+func PopulateZoneConfig(zonePresent, zonesPresent bool, zone, zones string, node *v1.Node, replicas int, allowedTopologies []v1.TopologySelectorTerm) (*VolumeZoneConfig, error) {
```

Comment:
Nr. of replicas is hardcoded to 1, so lot of code below is useless. I'd suggest to focus on the topology in this PR and then send a subsequent one with multi-zonal volumes.